### PR TITLE
6a styling sandbox

### PR DESCRIPTION
**Pull request by Scott-Borecki**
### Changes Implemented:
  - Updated and overhauled styling
  - Added icons (see the SVG files in `app/assets/icons/`; no I did not code the SVG files 😅 )
  - Completed the following stories:
    - #74 
    - #68 
    - #71 
    - #67 
    - #70 
    - #66 
    - #64 
    - #72 
    - #62 
    - #73 
    - #63 
    - #83 
    - #61 

### Quality Control Checklist:

  - [X] Code adheres to Rubocop styling.  If not, please clarify below:
  - [X] >90% SimpleCov test coverage.  If not, please clarify below:
  - [x] Last commit has passing CircleCI build

### Blockers (if applicable):
  - None

### Next Steps & Additional Notes:
  - Continue backfilling tests
  - Style new event page
  - Style welcome page
  - Style edit profile page
  - Let me know if you have any questions!  Would probably be good to run through together with a demo of the website.

### CODE REVIEW

File: app/assets/stylesheets/application.css.scss [hunk 1]
```diff
@@ -14,29 +14,49 @@
...
+
+$alert-text-color: #8e0613;
+
+$button-bg-color: #ffffff75;
+$link-color: #13525f;
+$navbar-bg-color: #99D3DFBD;
+$card-bg-color: #99D3DF50;
+$card-border-color: #88BBD695;
+$container-bg-color: #E9E9E9F5;
+$container-border-color: #CDCDCD95;
```
Comment (tvaroglu) [inline app/assets/stylesheets/application.css.scss:32]: Nice, set the vars above, and then just reference them in the class selectors below.

I had tried setting these directly within class selectors once, and got an 'invalid CSS error'.

Note to team on this!   Best practices for styling, as all we have to do is reset one of these vars up top (vs having to record every single class selector)

File: app/assets/stylesheets/application.css.scss [hunk 2]
```diff
@@ -45,69 +65,177 @@ footer {
...
 }
 
-.navbar-red {
-  background-color: #D4A69E95
+.card-display {
+  background: $card-bg-color;
+  min-width: 120px;
+  padding: 10px 20px 10px;
+  border: 4px solid $card-border-color;
+  border-radius: 8px 24px;
```
Comment (tvaroglu) [inline app/assets/stylesheets/application.css.scss:139]: Border radius for the win!!

File: app/assets/stylesheets/application.css.scss (hunk 1 above)
Comment (Scott-Borecki) [inline app/assets/stylesheets/application.css.scss:32]: FYI ... these are declared variables so we can quickly change colors if we want to adjust styling across the website.

File: app/controllers/events_controller.rb [hunk 3]
```diff
@@ -14,7 +14,7 @@ def create
 
   def destroy
     EventFacade.delete_event(event_destroy_params)
-    redirect_to dashboard_index_path
+    redirect_to URI(request.referer).path
```
Comment (tvaroglu) [inline app/controllers/events_controller.rb:17]: Cool syntax, will have to experiment with this going forward!

File: app/helpers/application_helper.rb [hunk 4]
```diff
@@ -1,2 +1,17 @@
...
+      file = asset.source.force_encoding("UTF-8")
+      doc = Nokogiri::HTML::DocumentFragment.parse file
+      svg = doc.at_css "svg"
+      svg["class"] = options[:class] if options[:class].present?
+    else
+      doc = "<!-- SVG #{filename} not found -->"
+    end
+
+    raw doc
+  end
```
Comment (Scott-Borecki) [inline app/helpers/application_helper.rb:16]: Check out this reference for where this helper method came from: https://blog.cloud66.com/using-svgs-in-a-rails-stack/

This method basically helps us display the SVG files in the views.

Comment (tvaroglu) [reply app/helpers/application_helper.rb:16]: Niceeee, thank you for posting source!

File: app/helpers/application_helper.rb [hunk 5]
```diff
@@ -1,2 +1,20 @@
 module ApplicationHelper
+  # This helper method allows us to display SVG images in the views.
+  # See this reference for an explanation of the helper method:
+  #   https://blog.cloud66.com/using-svgs-in-a-rails-stack/
```
Comment (tvaroglu) [inline app/helpers/application_helper.rb:4]: Thank you!!

File: app/poros/event.rb [hunk 6]
```diff
@@ -1,15 +1,22 @@
...
-    date.to_datetime.strftime('%A, %b %d, %Y %l:%M%P')
+  # TODO: Update method to handle different timezones
+  def format_date
+    DateTime.strptime(date_time, '%Y-%m-%dT%H:%M:%S.000').strftime('%A, %b %d, %Y %l:%M%P')
+  end
+
+  # TODO: Update method to handle different timezones
+  def format_date_short
+    DateTime.strptime(date_time, '%Y-%m-%dT%H:%M:%S.000').strftime('%a %m/%d/%y %l:%M%P')
   end
```
Comment (Scott-Borecki) [inline app/poros/event.rb:21]: We'll need to revisit this and how we are formatting the dates.  Right now, the methods assume that the incoming dates are in 000Z timezone.

File: app/poros/user.rb [hunk 7]
```diff
@@ -34,4 +34,13 @@ def self.goals_array
...
     ]
   end
+
+  def availability
+    availability = []
+    availability << 'Morning' if availability_morning
+    availability << 'Afternoon' if availability_afternoon
+    availability << 'Evening' if availability_evening
+
+    availability.join(', ')
```
Comment (tvaroglu) [inline app/poros/user.rb:44]: Love this!!

File: app/views/dashboard/index.html.erb [hunk 8]
```diff
@@ -1,86 +1,230 @@
...
+        <%= link_to 'View Profile', profile_path(current_user.id), method: :get, class: "btn btn-sm btn-info button-bg btn-outline-dark" %>
+        <%= link_to 'Edit Profile', edit_profile_path(current_user.id), method: :get, class: "btn btn-sm btn-info button-bg btn-outline-dark" %>
+      </div>
+
+      <div class="col-lg fit-content">
+        <ul class='no-bullets'>
+          <li>
+            <%= embedded_svg('icons/basic_geolocalize-01.svg', class: 'icon') %>
+            &nbsp;<%= current_user.zip_code %>
+          </li>
```
Comment (Scott-Borecki) [inline app/views/dashboard/index.html.erb:21]: FYI ... this `embedded_svg` method is the method referenced in the `ApplicationHelper` module for displaying `.svg` image files.

Here is the reference again: https://blog.cloud66.com/using-svgs-in-a-rails-stack/

File: app/poros/event.rb [hunk 9]
```diff
@@ -1,15 +1,24 @@
...
     @gym_membership_id = data[:attributes][:gym_membership_id]
-    @date_time         = format_date(data[:attributes][:date_time])
+    @date_time         = data[:attributes][:date_time]
     @activity          = data[:attributes][:activity]
   end
 
-  def format_date(date)
-    date.to_datetime.strftime('%A, %b %d, %Y %l:%M%P')
+  # TODO: Update method to handle different timezones
+  # Returns the date in the format of 'Wednesday, Sep 22, 2021  9:41pm'
```
Comment (tvaroglu) [inline app/poros/event.rb:14]: Great clarity - thank you!

File: spec/features/users/dashboard/new_user_spec.rb [hunk 10]
```diff
@@ -20,23 +20,23 @@
...
+            expect(page).to have_content('You have no friends.')
           end
         end
       end
 
       context "when I don't have any gyms" do
         it 'displays "You currently are not a member of any gyms."', :vcr do
           within '#gyms' do
-            expect(page).to have_content('You currently are not a member of any gyms.')
+            expect(page).to have_content('You are not a member of any gym.')
```
Comment (tvaroglu) [inline spec/features/users/dashboard/new_user_spec.rb:31]: Way more readable, good call UX-wise

**Review by tvaroglu — APPROVED** [review]
Amazing work, thank you so much for being so thorough, and cleaning up the css file along the way!  This will be extremely helpful moving forward to add additional features, with a consistent look and feel!!

**Comment by Scott-Borecki** [comment]
Update testing to improve test coverage to 95.45%.

File: app/poros/event.rb (hunk 6 above)
Comment (Scott-Borecki) [reply app/poros/event.rb]: I updated this in a later commit here

File: app/assets/stylesheets/application.css.scss (hunk 1 above)
Comment (Scott-Borecki) [reply app/assets/stylesheets/application.css.scss:32]: Here are the Sass docs on variables: https://sass-lang.com/documentation/variables

File: app/controllers/gym_memberships_controller.rb [hunk 11]
```diff
@@ -1,12 +1,12 @@
...
   def create
     GymMembershipFacade.create_gym_membership(gym_membership_params)
-    redirect_to dashboard_index_path
+    redirect_to URI(request.referer).path
   end
 
   def destroy
     GymMembershipFacade.delete_gym_membership(gym_membership_destroy_params)
-    redirect_to dashboard_index_path
+    redirect_to URI(request.referer).path
```
Comment (gunnarrunner) [inline app/controllers/gym_memberships_controller.rb:9]: Nice change. Looks professional and is more intricate.

File: app/views/dashboard/index.html.erb (hunk 8 above)
Comment (gunnarrunner) [reply app/views/dashboard/index.html.erb:21]: Very fun to see how this is utilized in the views. Thank you!

**Review by gunnarrunner — APPROVED** [review]
LGTM!

**Pull request merged**